ecosystem/nft-metadata-crawler/nft-metadata-crawler-utils/src/gcs.rs

```diff
@@ -1,23 +1,92 @@
 // Copyright © Aptos Foundation
 
 use image::ImageFormat;
+use reqwest::{
+    header::{self, HeaderMap},
+    Client,
+};
 use serde_json::Value;
 
+/// Writes JSON Value to GCS
 pub async fn write_json_to_gcs(
-    _token: String,
-    _bucket: String,
-    _id: String,
-    _json: Value,
+    token: String,
+    bucket: String,
+    id: String,
+    json: Value,
 ) -> anyhow::Result<String> {
-    todo!();
+    let client = Client::new();
+    let filename = format!("json_{}.json", id);
+    let url = format!(
+        "https://storage.googleapis.com/upload/storage/v1/b/{}/o?uploadType=media&name={}",
+        bucket, filename
+    );
+    let json_string = json.to_string();
+
+    let res = client
+        .post(url)
+        .bearer_auth(token)
+        .header("Content-Type", "application/json")
+        .body(json_string)
+        .send()
+        .await?;
+
+    match res.status().as_u16() {
+        200..=299 => Ok(filename),
+        _ => {
+            let text = res.text().await?;
+            Err(anyhow::anyhow!("Error saving JSON to GCS {}", text))
+        },
+    }
 }
 
+/// Infers file type and writes image to GCS
 pub async fn write_image_to_gcs(
-    _token: String,
-    _img_format: ImageFormat,
-    _bucket: String,
-    _id: String,
-    _buffer: Vec<u8>,
+    token: String,
+    img_format: ImageFormat,
+    bucket: String,
+    id: String,
+    buffer: Vec<u8>,
 ) -> anyhow::Result<String> {
-    todo!();
+    let client = Client::new();
+    let mut headers = HeaderMap::new();
+
+    let extension = match img_format {
+        ImageFormat::Gif | ImageFormat::Avif => img_format
+            .extensions_str()
+            .last()
+            .unwrap_or(&"gif")
+            .to_string(),
+        _ => "jpeg".to_string(),
```

Comment:
why no png here?

Reply:
Converting all PNGs to JPEGs so they are smaller 🙂